dotnet/src/Connectors/Connectors.Amazon/Bedrock/Core/Clients/BedrockChatCompletionClient.cs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) Microsoft. All rights reserved.
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Connectors.Amazon.Core.Requests;
+using Connectors.Amazon.Core.Responses;
+using Connectors.Amazon.Models;
+using Connectors.Amazon.Models.AI21;
+using Connectors.Amazon.Models.Amazon;
+using Connectors.Amazon.Models.Anthropic;
+using Connectors.Amazon.Models.Cohere;
+using Connectors.Amazon.Models.Meta;
+using Connectors.Amazon.Models.Mistral;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel.Connectors.Amazon.Core;
+/// <summary>
+/// Represents a client for interacting with the chat completion through Bedrock.
+/// </summary>
+/// <typeparam name="TRequest"> Request object which is an IChatCompletionRequest. </typeparam>
+/// <typeparam name="TResponse"> Response object which is an IChatCompletionResponse. </typeparam>
+public class BedrockChatCompletionClient<TRequest, TResponse>
+    where TRequest : IChatCompletionRequest
+    where TResponse : IChatCompletionResponse
+{
+    private readonly string _modelId;
+    private readonly string _modelProvider;
+    private readonly IAmazonBedrockRuntime _bedrockApi;
+    private readonly IBedrockModelIOService<IChatCompletionRequest, IChatCompletionResponse> _ioService;
+    private readonly Uri _chatGenerationEndpoint;
+
+    /// <summary>
+    /// Builds the client object and registers the model input-output service given the user's passed in model ID.
+    /// </summary>
+    /// <param name="modelId"></param>
+    /// <param name="bedrockApi"></param>
+    /// <exception cref="ArgumentException"></exception>
+    public BedrockChatCompletionClient(string modelId, IAmazonBedrockRuntime bedrockApi)
+    {
+        this._modelId = modelId;
+        this._bedrockApi = bedrockApi;
+        this._chatGenerationEndpoint = new Uri("https://bedrock-runtime.us-east-1.amazonaws.com");
+        int periodIndex = modelId.IndexOf('.'); //modelId looks like "amazon.titan-embed-text-v1:0"
+        string modelProvider = periodIndex >= 0 ? modelId.Substring(0, periodIndex) : modelId;
+        this._modelProvider = modelProvider;
+        switch (modelProvider)
+        {
+            case "ai21":
+                if (modelId.Contains("jamba"))
```

Comment:
Optional: Rather than .Contains, you could consider something more specific like StartsWith, or make the string you're looking for more complete. I'm just concerned if say a new model comes out called "Better-jamba", this will match it even if we haven't added the code needed for that model.